### PR TITLE
feat: add routing and navigation

### PR DESCRIPTION
**Pull request by timoturkki**
- adds routing
- adds navigation component with three links
- adds placeholder components for different views
- changes so tests are not watching by default
- adds theme for material ui (only one colour, needs to be worked on)

Probably would be nice to customise the navigation a bit later, added just some default components. Also needs still the search input that we discussed

### CODE REVIEW

File: src/ui/components/navigation/Navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import AppBar from '@material-ui/core/AppBar';
+import Toolbar from '@material-ui/core/Toolbar';
+import Button from '@material-ui/core/Button';
+
+const Navigation: FunctionComponent = () => {
+  return (
+    <AppBar color="primary" position="static">
+      <Toolbar className="nav-container">
+        <Button color="inherit" component={Link} to="/">
```
Comment (Boochoo) [inline src/ui/components/navigation/Navigation.tsx:13]: Those 3 buttons can be one reusable component which takes those attributes as props. We can use the component in here with a prop that receives array of objects eg: `[{name: ''home, to: '/'}, {name: ''login, to: '/login'}, {name: ''signup, to: '/signup'}]` and loop to display those. probably basic but can be useful when we expand links through out the pages

File: src/ui/components/navigation/Navigation.test.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { render } from '@testing-library/react';
+import Navigation from './Navigation';
+
+import { BrowserRouter as Router } from 'react-router-dom';
```
[the remark this answers is not in the document]
Comment (timoturkki) [reply src/ui/components/navigation/Navigation.test.tsx:3]: yep true, changed the order. I think we should be able to enforce this with eslint rules also, have to look at it separetely

File: src/ui/components/router-links/RouterLinks.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  links: NavigationLink[];
+}
+
+const RouterLinks: FunctionComponent<RouterLinksProps> = ({ links }) => {
+  return (
+    <React.Fragment>
+      {links.map((link) => (
+        <Button
+          color="inherit"
+          key={'router-link' + shortid.generate()}
```
Comment (Boochoo) [inline src/ui/components/router-links/RouterLinks.tsx:19]: awesome

Comment (Boochoo) [reply src/ui/components/router-links/RouterLinks.tsx:19]: not sure how big the lib is, is it possible to destruct generate instead of *?

Comment (Boochoo) [reply src/ui/components/router-links/RouterLinks.tsx:19]: No biggie. You don't have to make the changes, but noticed that we can also destruct Fragment from react. name and target from links array as well. `links.map((({name, target}))`

Comment (timoturkki) [reply src/ui/components/router-links/RouterLinks.tsx]: done 👍

**Pull request merged**